pkg/ranchhand/processhosts.go

```diff
@@ -263,3 +246,75 @@ func constrainDockerVersion(client *ssh.Client) error {
 
 	return nil
 }
+
+// install docker onto a new system and mark the operation as complete thereafter
+func installDocker(client *ssh.Client, osInfo *osi.Info) error {
+	if err := createRemoteDir(client, remoteStateDir); err != nil {
+		return errors.Wrap(err, "cannot create remote state dir")
+	}
+
+	indicator := filepath.Join(remoteStateDir, "docker-installed")
+	if _, err := client.ExecuteCmd(fmt.Sprintf("test -f %s", indicator)); err == nil {
+		return nil
+	}
+
+	cmds := append(dockerInstallCmds[osInfo.ID], "sudo usermod -aG docker $USER")
+	log.Infof("installing docker on host [%s]", client.RemoteAddr())
+	if _, err := client.ExecuteCmd(strings.Join(cmds, " && ")); err != nil {
+		return errors.Wrap(err, "docker install failed")
+	}
+
+	err := retry.Retry(
+		func(attempt uint) (err error) {
+			if _, err = client.ExecuteCmd("sudo docker version"); err != nil {
+				log.Warnf("attempt [%d] to verify docker is running failed on host [%s]", attempt, client.RemoteAddr())
+			}
+			return err
+		},
+		strategy.Wait(10*time.Second),
+		strategy.Limit(12),
+	)
+	if err != nil {
+		return errors.Wrap(err, "unable to verify docker install")
+	}
+
+	_, err = client.ExecuteCmd(fmt.Sprintf("sudo touch %s", indicator))
+	return errors.Wrap(err, "cannot mark docker install complete")
+}
+
+// install static k8s config files
+func installK8sConfigs(client *ssh.Client) error {
+	if err := createRemoteDir(client, "/etc/kubernetes"); err != nil {
+		return errors.Wrap(err, "cannot create k8s cfg dir")
+	}
+	if err := createStaticCfg(client, k8sAuditCfg, "/etc/kubernetes/audit.yaml"); err != nil {
+		return errors.Wrap(err, "cannot create k8s audit logging cfg")
+	}
+	if err := createStaticCfg(client, k8sAdmissionCfg, "/etc/kubernetes/admission.yaml"); err != nil {
+		return errors.Wrap(err, "cannot create k8s admission policy cfg")
+	}
+	if err := createStaticCfg(client, k8sEventCfg, "/etc/kubernetes/event.yaml"); err != nil {
+		return errors.Wrap(err, "cannot create k8s event limiting cfg")
+	}
+
+	return nil
+}
+
+func createStaticCfg(client *ssh.Client, contents, filename string) error {
+	if _, cerr := client.ExecuteCmd(fmt.Sprintf("test -f %s", filename)); cerr != nil {
+		genCmds := fmt.Sprintf("echo -e %[1]q | sudo tee %[2]s && sudo chown root:root %[2]s && sudo chmod 0600 %[2]s", contents, filename)
+		if _, err := client.ExecuteCmd(genCmds); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func createRemoteDir(client *ssh.Client, name string) error {
+	if _, cerr := client.ExecuteCmd(fmt.Sprintf("test -d %s", name)); cerr != nil {
+		if _, err := client.ExecuteCmd(fmt.Sprintf("sudo mkdir -p %s", name)); err != nil {
```

Comment:
Why not `test -d %s || mkdir -p`? Or even just `mkdir -p`?

Reply:
`test -d %s || mkdir -p` is certainly better than 2 remote calls. I just tested `mkdir -p` and read the man page. I think that's the best route.